addons/mail/tests/test_link_preview.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from functools import partial
+
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.mail.tests.common import MailCommon
+from odoo.tests import tagged
+from unittest.mock import patch
+import requests
+
+mail_channel_new_test_user = partial(mail_new_test_user, context={'mail_channel_nosubscribe': False})
+
+
+def _patched_get_html(*args, **kwargs):
+    response = requests.Response()
+    response.status_code = 200
+    response._content = b"""
+    <html>
+    <head>
+    <meta property="og:title" content="Test title">
+    <meta property="og:description" content="Test description">
+    </head>
+    </html>
+    """
+    response.headers["Content-Type"] = 'text/html'
+    return response
+
+
+@tagged('LinkPreview')
+class TestLinkPreview(MailCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestLinkPreview, cls).setUpClass()
```

Comment:
```suggestion
        super().setUpClass()
```
been a while since we last supported Python 2.